app/views/sign_up/recovery_codes/_recovery_page_controller.js.erb

```diff
@@ -0,0 +1,96 @@
+var modalSelector = '#personal-key-confirm';
+var modal = new LoginGov.Modal({el: modalSelector });
+
+var recoveryWords = [].slice.call(document.querySelectorAll('[data-recovery]'));
+var formEl = document.getElementById('confirm-key');
+var inputs = [].slice.call(formEl.elements).filter(function(el) { return el.type === 'text'});
+var modalTrigger = document.querySelector('[data-toggle="modal"]');
+var modalDismiss = document.querySelector('[data-dismiss="personal-key-confirm"]')
+
+var reminderEl = document.getElementById('recovery-code-reminder');
+
+var isInvalidForm = false;
+var alertFallback = '<div class="alert alert-alert" role="alert" id="recovery-code-alert">' +
+                    "<%= t('users.recovery_code.confirmation_error') %>" +
```

Comment:
Definitely agree that just having a string of HTML is bad, and I should have paid more attention before pushing it and putting the `review` flag on this PR.

Just to be really specific, I was asking which of the two options (hiding html vs document.createElement) you consider to be a 'best practice' in a rails app.